lib/queuePopulator/LogReader.js

```diff
@@ -365,7 +372,14 @@ class LogReader {
                 batchState.publishedEntries[topic] = topicEntries;
                 return done();
             });
-        }, done);
+        }, () => {
+            this._extensions.forEach(ext => {
```

Comment:
Should this `forEach` instead be asynchronous?

Reply:
Ya thats a good idea 👍  this and in `MetricsProducer.publishMetrics` https://github.com/scality/backbeat/pull/125/files/aefeaf3118bfc397794c9255ca249a339f765670#diff-5062767101f20bda5a3063b13a8d46adR52
Both look like good opportunities for async calls